Pg 18 compatibility

rapimo opened the pull request:
@sschmidtadjust requested to ensure pg17 compatibility 
ref https://adjustcom.atlassian.net/browse/DB-1822

rapimo committed:
add pg18 and pg17 compatibility

pg 17 and 18 added additional parameters to create_foreignscan_path
namely `int disabled_nodes` (pg18) and `List *fdw_restrictinfo` (pg17)
which we both don't use for the moment.

rename static json_categorize_type to kafka_json_categorize_type

in pg 17 `json_categorize_type` became part of the public api while
we could use it but still need a static version for older versions
anyway, so we simply maintain our own version for now.

similar datum_to_json is now public with different signature

remove tuplestore_donestoring

actually `tuplestore_donestoring` was never needed and got removed in pg17
ref https://github.com/postgres/postgres/commit/75680c3d805e2323cd437ac567f0677fdfc7b680
    https://github.com/postgres/postgres/commit/dd04e958c8b03c0f0512497651678c7816af3198

add alternative explain output for kafka_test.out

some EXPLAIN output changed in pg17 and pg18 and went into
alternative output files

avoid shadow variable errors

if compiled with -Wshadow=compatible-local we might see warning within the
PG_TRY CATCH machinery, this can be avoided by using variable suffixes introduced in pg16

rapimo committed:
use INT64_FORMAT to be platform independent

the `INT64_FORMAT` macro takes care for the right formatspecifier
of an int64 (typically long long or long)

rapimo commented:
@PerikAdjust maybe you can have a look into the build farm? Not sure where it fails there and if it ever worked properly.

PerikAdjust commented:
I think you can first remove [ci_dockerfile.yml](https://github.com/adjust/kafka_fdw/blob/master/.github/workflows/ci_dockerfile.yml) we don't need it if we have ci.yml (the one that you update). 
Because the failing checks come from the previous workflows I think

rapimo committed:
Makefile makeover

rapimo committed:
use new --bootstrap-server option

Starting with Apache Kafka 2.2.0, a new argument was introduced:
```
--bootstrap-server <broker:port>
```

which connects directly to Kafka brokers instead of ZooKeeper.

Then, from Kafka 3.0.0 onward, ZooKeeper-based options (`--zookeeper`)
became deprecated.
And from Kafka 4.0.0, ZooKeeper will be completely removed.
